mods/bones/init.lua

```diff
@@ -211,81 +245,100 @@ minetest.register_on_dieplayer(function(player)
 		return
 	end
 
-	local player_inv = player:get_inventory()
-	if is_all_empty(player_inv) then
-		minetest.log("action", player_name .. " dies at " .. pos_string ..
-			". No bones placed")
-		if bones_position_message then
-			minetest.chat_send_player(player_name, S("@1 died at @2.", player_name, pos_string))
-		end
-		return
-	end
+	for _, fun in ipairs(dead_player_callbacks) do
```

Comment:
I feel it should be possible to refactor this loop by splitting it up into functions:

* A function to collect the items to be added to the inventory in a list. This function would run the callbacks and append all the items to a list.
* A function to find a suitable position for a bones node.
* A function to set and populate the inventory of set node.

Perhaps the latter two functions should be only one. You might also want to inline these, but the important thing is that I'd like to have this as a clear-cut three stage process:

1. Collect items to be placed in the inventory
2. Find a position to place the block
3. Place the block and populate the inventory

in order to get rid of the very stateful loop with high cyclomatic complexity